protovalidate/validator.py

```diff
@@ -98,15 +101,25 @@ class ValidationError(ValueError):
     An error raised when a message fails to validate.
     """
 
-    violations: validate_pb2.Violations
+    _violations: list[Violation]
 
-    def __init__(self, msg: str, violations: validate_pb2.Violations):
+    def __init__(self, msg: str, violations: list[validate_pb2.Violations]):
```

Comment:
something about the types here doesn't quite match up, we're passing in the list of proto messages (which is returned raw from `def violations`) + `_violations` is typed as `list[Violation]`. I think they all need to match up?

Reply:
Yeah, you're right. It's unfortunate how easy it is to have extremely wrong types pass type checking. Oh well.

Should be fixed now. (It's a little awkward because Mypy doesn't support "aliasing" a type. Pylance/etc. don't seem to mind.)